components/inputs/input-search.js

```diff
@@ -0,0 +1,238 @@
+				}
+				:host([dir="rtl"]) d2l-button-icon {
+					left: 0.3rem;
+					right: auto;
+				}
+			`
+		];
+	}
+
+	static get resources() {
+		return {
+			'ar': { 'search': 'بحث', 'search.clear': 'مسح البحث' },
+			'en': { 'search': 'Search', 'search.clear': 'Clear Search' },
+			'es': { 'search': 'Buscar', 'search.clear': 'Borrar búsqueda' },
+			'fr': { 'search': 'Rechercher', 'search.clear': 'Effacer la recherche' },
+			'ja': { 'search': '検索', 'search.clear': '検索のクリア' },
+			'ko': { 'search': '검색', 'search.clear': '검색 지우기' },
+			'nl': { 'search': 'Zoeken', 'search.clear': 'Zoekopdracht wissen' },
+			'pt': { 'search': 'Pesquisar', 'search.clear': 'Limpar Pesquisa' },
+			'sv': { 'search': 'Sökning', 'search.clear': 'Rensa sökning' },
+			'tr': { 'search': 'Ara', 'search.clear': 'Aramayı Temizle' },
+			'zh': { 'search': '搜索', 'search.clear': '清除搜索' },
+			'zh-tw': { 'search': '搜尋', 'search.clear': '清除搜尋' }
+		};
+	}
+
+	constructor() {
+		super();
+		this._focussed = false;
+		this._hovered = false;
+		this._lastSearchValue = '';
+		this.disabled = false;
+		this.noClear = false;
+		this.value = '';
+	}
+
+	get lastSearchValue() { return this._lastSearchValue; }
+	set lastSearchValue(val) {}
+	_setLastSearchValue(val) {
+		const oldVal = this._lastSearchValue;
+		this._lastSearchValue = val;
+		this.requestUpdate('lastSearchValue', oldVal);
+	}
+
+	render() {
+		requestAnimationFrame(
+			() => this.dispatchEvent(new CustomEvent('d2l-input-search-render', {
+				bubbles: false,
+				composed: false
+			}))
+		);
+		const inputClasses = {
+			'd2l-input': true,
+			'd2l-input-focus': !this.disabled && (this._focussed || this._hovered)
+		};
+		const showSearch = this._computeShowSearch();
+		return html`
+			<div class="d2l-input-search-container"
+				@mouseout="${this._handleMouseLeave}"
+				@mouseover="${this._handleMouseEnter}">
+				<input
+					aria-label="${ifDefined(this.label)}"
+					class="${classMap(inputClasses)}"
+					?disabled="${this.disabled}"
+					@input="${this._handleInput}"
+					@keypress="${this._handleInputKeyPress}"
+					maxlength="${ifDefined(this.maxlength)}"
+					placeholder="${ifDefined(this.placeholder)}"
+					type="search"
+					.value="${this.value}">
+				<d2l-button-icon
+					class="d2l-input-search-search"
+					?disabled="${this.disabled}"
+					?hidden="${!showSearch}"
+					icon="tier1:search"
+					@click="${this.search}"
+					text="${this.localize('search')}"></d2l-button-icon>
+				<d2l-button-icon
+					class="d2l-input-search-clear"
+					@click="${this._handleClearClick}"
+					?disabled="${this.disabled}"
+					?hidden="${showSearch}"
+					icon="tier1:close-default"
+					text="${this.localize('search.clear')}"></d2l-button-icon>
+			</div>
+		`;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (this.value !== undefined && this.value !== null) {
+			this._setLastSearchValue(this.value);
+		}
+		this.addEventListener('blur', this._handleBlur);
```

Comment:
I think in the original code, these handlers were wire-up with `capture: true`. I am wondering if focus management is still working? 'blur' doesn't bubble, so it looks like its handler would only get invoked when the outer host loses focus, not when the internal focusable elements lose focus.

Reply:
This was the result of some trial & error -- it is still working yes, even without the capture. 

An inline `@focus` listener on the container didn't work since as you mentioned it doesn't bubble. Listeners on the input and both buttons also didn't work since the focus/blur events actually happen on the underlying `<button>` elements, which are buried.

The focus event now only fires when focus goes into the component and blur only fires once when it leaves, as you suspected. This works, but it also means I don't need to check the `relatedTarget` anymore. I'm going to look at that code more closely and remove if possible.

Reply:
I've removed the `relatedTarget` code.